chain/chain.go

```diff
@@ -38,26 +38,47 @@ type RPCClient struct {
 	quitMtx sync.Mutex
 }
 
-// NewRPCClient creates a client connection to the server described by the
+// NewRPCClient creates a direct client connection to the server described by the
 // connect string.  If disableTLS is false, the remote RPC certificate must be
 // provided in the certs slice.  The connection is not established immediately,
 // but must be done using the Start method.  If the remote server does not
 // operate on the same decred network as described by the passed chain
 // parameters, the connection will be disconnected.
 func NewRPCClient(chainParams *chaincfg.Params, connect, user, pass string, certs []byte,
 	disableTLS bool) (*RPCClient, error) {
+	return NewClient(chainParams, connect, user, pass,
+		certs, disableTLS, "", "", "")
+}
+
+// NewClient creates a client connection to the server described by the
+// connect string.  If disableTLS is false, the remote RPC certificate must be
+// provided in the certs slice.  The connection is not established immediately,
+// but must be done using the Start method.  If the remote server does not
+// operate on the same decred network as described by the passed chain
+// parameters, the connection will be disconnected. If proxypath is not empty,
+// it will be used to connect.
+func NewClient(chainParams *chaincfg.Params, connect, user, pass string, certs []byte,
```

Comment:
If NewClientConfig is exported, I don't see any reason to also include this.

Reply:
One reason is that any caller of NewRPCClientConfig will need to import dcrrpcclient package, and any user of NewRPCClient can simply update to using SOCKS without changing too much code.

Reply:
It's a transient dependency anyways, making it an explicit import doesn't change anything and adds less cruft to the public API.